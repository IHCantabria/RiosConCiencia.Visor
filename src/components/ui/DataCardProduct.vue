<template>
  <div class="overviewcard">
    <div class="overviewcard__description">
      <vs-list class="additional-Info">
        <vs-list-item class="additional-Info__Icon">
          <i
            :class="[openLicense ? 'fas fa-unlock' : 'fas fa-lock']"
            :title="licenseType"
          ></i>
        </vs-list-item>
        <vs-list-item class="additional-Info__Icon">
          <a :href="catalogUrl" title="Go to Data Catalog" target="_blank">
            <i class="fas fa-folder"></i>
          </a>
        </vs-list-item>
        <vs-list-item title="Type of Product"
          ><span>{{ typeProduct }}</span></vs-list-item
        >
        <vs-list-item title="Source Origin"
          ><a :href="sourceUrl" title="Go to source Url" target="_blank">
            <span>{{ sourceName }}</span>
          </a></vs-list-item
        >
        <vs-list-item title="Source Ambit"
          ><span>{{ sourceType }}</span></vs-list-item
        >
      </vs-list>
      <vs-list class="variables-Info">
        <vs-list-header
          icon="info"
          title="Variables Info"
          color="primary"
        ></vs-list-header>
        <vs-list-item
          v-for="variable in variables"
          :key="variable.id"
          :title="variable.longName"
        >
          <span><strong>Short Name</strong>: {{ variable.shortName }}</span>
          <span><strong>Alias</strong>: {{ variable.alias }}</span>
          <span
            ><strong>Units</strong>: {{ variable.units }}</span
          ></vs-list-item
        >
      </vs-list>
    </div>
  </div>
</template>
<script>
export default {
  props: {
    openLicense: {
      type: Boolean,
      required: false,
      default: () => {
        return false;
      }
    },
    licenseType: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    catalogUrl: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    typeProduct: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    sourceName: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    sourceType: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    sourceUrl: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    variables: {
      type: Array,
      required: false,
      default: () => {
        return [];
      }
    }
  }
};
</script>
<style lang="scss" scoped>
.overviewcard {
  border-top: solid $color-primary;
  border-bottom: solid $color-primary;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  background-color: $color-very-light-grey;

  &__description {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
}
</style>
