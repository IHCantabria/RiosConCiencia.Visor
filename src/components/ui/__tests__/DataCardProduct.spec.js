import { shallowMount, createLocalVue } from "@vue/test-utils";
import DataCardProduct from "../DataCardProduct.vue";
import Vuesax from "vuesax";
import merge from "lodash.merge";

const localVue = createLocalVue();
localVue.use(Vuesax);

function createWrapper(overrides) {
  const defaultMountingOptions = { localVue };
  return shallowMount(
    DataCardProduct,
    merge(defaultMountingOptions, overrides)
  );
}

describe("dataCardProduct.vue", () => {
  //SNAPSHOT TESTS
  it("renders correctly default values", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly with full info", () => {
    expect.assertions(1);
    const wrapper = createWrapper({
      propsData: {
        openLicense: true,
        licenseType: "open",
        catalogUrl: "url test",
        typeProduct: "type test",
        sourceName: "source test",
        sourceType: "source type test",
        sourceUrl: "url source test",
        variables: [
          {
            id: 1,
            longName: "long name test 1",
            shortName: "short name test 1",
            alias: "alias test 1",
            units: "m"
          },
          {
            id: 2,
            longName: "long name test 2",
            shortName: "short name test 2",
            alias: "alias test 2",
            units: "s"
          }
        ]
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
