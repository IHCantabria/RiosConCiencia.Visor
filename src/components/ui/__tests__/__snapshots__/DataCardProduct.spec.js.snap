// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataCardProduct.vue renders correctly default values 1`] = `
<div
  class="overviewcard"
>
  <div
    class="overviewcard__description"
  >
    <vs-list-stub
      class="additional-Info"
    >
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <i
          class="fas fa-lock"
          title=""
        />
      </vs-list-item-stub>
       
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <a
          href=""
          target="_blank"
          title="Go to Data Catalog"
        >
          <i
            class="fas fa-folder"
          />
        </a>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Type of Product"
      >
        <span>
          
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Source Origin"
      >
        <a
          href=""
          target="_blank"
          title="Go to source Url"
        >
          <span>
            
          </span>
        </a>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Source Ambit"
      >
        <span>
          
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
     
    <vs-list-stub
      class="variables-Info"
    >
      <vs-list-header-stub
        color="primary"
        icon="info"
        iconpack="material-icons"
        title="Variables Info"
      />
       
    </vs-list-stub>
  </div>
</div>
`;

exports[`dataCardProduct.vue renders correctly with full info 1`] = `
<div
  class="overviewcard"
>
  <div
    class="overviewcard__description"
  >
    <vs-list-stub
      class="additional-Info"
    >
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <i
          class="fas fa-unlock"
          title="open"
        />
      </vs-list-item-stub>
       
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <a
          href="url test"
          target="_blank"
          title="Go to Data Catalog"
        >
          <i
            class="fas fa-folder"
          />
        </a>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Type of Product"
      >
        <span>
          type test
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Source Origin"
      >
        <a
          href="url source test"
          target="_blank"
          title="Go to source Url"
        >
          <span>
            source test
          </span>
        </a>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Source Ambit"
      >
        <span>
          source type test
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
     
    <vs-list-stub
      class="variables-Info"
    >
      <vs-list-header-stub
        color="primary"
        icon="info"
        iconpack="material-icons"
        title="Variables Info"
      />
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="long name test 1"
      >
        <span>
          <strong>
            Short Name
          </strong>
          : short name test 1
        </span>
         
        <span>
          <strong>
            Alias
          </strong>
          : alias test 1
        </span>
         
        <span>
          <strong>
            Units
          </strong>
          : m
        </span>
      </vs-list-item-stub>
      <vs-list-item-stub
        iconpack="material-icons"
        title="long name test 2"
      >
        <span>
          <strong>
            Short Name
          </strong>
          : short name test 2
        </span>
         
        <span>
          <strong>
            Alias
          </strong>
          : alias test 2
        </span>
         
        <span>
          <strong>
            Units
          </strong>
          : s
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
  </div>
</div>
`;
